stdlib/public/core/Sendable.swift

```diff
@@ -10,13 +10,103 @@
 //
 //===----------------------------------------------------------------------===//
 
-/// The Sendable protocol indicates that value of the given type can
-/// be safely used in concurrent code.
+/// A type whose values can safely be passed across concurrency domains by copying.
+///
+/// You can safely pass values of a sendable type
+/// from one concurrency domains to another ---
+/// for example, you can pass a sendable value as the argument
+/// when calling an actor's methods.
+/// All of the following can be marked as sendable:
+///
+/// - Value types
+/// - References to immutable reference types
+/// - Reference types that internally manage access to their state
+/// - Functions and closures
+///
+/// Although this protocol doesn't have any required methods or properties,
+/// it does have semantic requirements that are enforced at compile time.
+/// These requirements are listed in the sections below.
+/// Conformance to `Sendable` must be declared
+/// in the same file as the type's declaration.
+///
+/// To declare conformance to `Sendable` without any compiler enforcement,
+/// write `@unchecked Sendable`.
+/// You are responsible for the correctness of unchecked sendable types.
+/// Unchecked conformance to `Sendable` also disables enforcement
+/// of the rule that conformance must be in the same file.
+///
+/// For information about the language-level concurrency model that `Task` is part of,
+/// see [Concurrency][concurrency] in [The Swift Programming Language][tspl].
+///
+/// [concurrency]: https://docs.swift.org/swift-book/LanguageGuide/Concurrency.html
+/// [tspl]: https://docs.swift.org/swift-book/
+///
+/// ### Sendable Structures and Enumerations
+///
+/// To satisfy the requirements of the `Sendable` protocol,
+/// an enumeration or structure must have only sendable
+/// members and associated values.
+/// In some cases, structures and enumerations
+/// that satisfy the requirements implicitly conform to `Sendable`:
+///
+/// -   Frozen structures and enumerations
+///
+/// -   Structures and enumerations
+///     that aren't public and aren't marked `@usableFromInline`.
+///
+/// Otherwise, you need to declare conformance to `Sendable` explicitly.
+///
+/// ### Sendable Actors and Classes
+///
+/// All actor types implicitly conform to `Sendable`.
```

Comment:
I recommend leading with a sentence or two stating why we get implicit Sendable conformances, because it'll be informative and help with the "classes" section. Here's one idea:

```suggestion
/// Actor types are reference types where each instance is its own concurrency domain. The actor ensures that all accesses to its mutable state are performed sequentially. Therefore, all actor types implicitly conform to `Sendable`.
```

Reply:
Added in 32e08250685b69fdc44c85b33fa632279f032e5f:

> All actor types implicitly conform to `Sendable` because actors ensure that all access to their mutable state is performed sequentially.